R/install-CRAN.r

```diff
@@ -0,0 +1,58 @@
+#' Attempts to install a package from CRAN.
+#'
+#' This function is vectorised on \code{pkgs} so you can install multiple
+#' packages in a single command.
+#'
+#' @param pkgs Character vector of packages to install.
+#' @inheritParams package_deps
+#' @export
+#' @family package installation
+#' @examples
+#' \dontrun{
+#' install_CRAN("ggplot2")
+#' install_CRAN(c("httpuv", "shiny")
+#' }
+install_CRAN <- function(pkgs, repos = getOption("repos"), type = getOption("pkgType"), ..., quiet = FALSE) {
+
+  remotes <- lapply(pkgs, CRAN_remote, repos = repos, type = type)
+
+  install_remotes(remotes, quiet = quiet, ...)
+}
+
+CRAN_remote <- function(pkg, repos, type) {
+
+  remote("CRAN",
+    name = pkg,
+    repos = repos,
+    pkg_type = type)
+}
+
+
+#' @export
+remote_download.CRAN_remote <- function(x, quiet = FALSE) {
+  dest_dir <- tempdir()
+  download.packages(x$name, destdir = dest_dir, repos = x$repos, type = x$pkg_type)[1, 2]
```

Comment:
Dependencies are all done in `install()` itself which calls `install_deps()` so we don't have to worry about it here.